.tekton/tasks/sync-argocd-app.yaml

```diff
@@ -0,0 +1,69 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: app-sync
+spec:
+  description: >-
+    This task is for syncing the staging cluster argocd application
+  workspaces:
+   - name: argocd-config
+   - name: infra-deployment-git
+  steps:
+    - name: argocd-app-sync
+      image: quay.io/devtools_gitops/test_image:4.0.5
+      workingDir: $(workspaces.argocd-config.path)
+      script: |
+        #!/usr/bin/env python3
+        import subprocess
+        import re
+        import os
+
+        cmd = "pip3 install pyyaml"
+        output = subprocess.Popen(cmd.split(), stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        stdout, stderr = output.communicate()
+
+        # Load applications file
+        with open('$(workspaces.infra-deployment-git.path)/staging-app.yaml') as f:
+          import yaml
+          app = yaml.safe_load(f)
+
+        keys_pattern = r"\b(" + "|".join(re.escape(key) for key in app.keys()) + r")\b"
+
+        # Print the list of file paths
+        matches = []
+        for file in open('$(workspaces.infra-deployment-git.path)/updated_files.txt').readlines():
+          file = file.strip()
+          path_matches = re.findall(keys_pattern, file)
+          matches.extend(path_matches)
+
+        # Find matches in pull request files
+        unique_matches = set(matches)
+
+        # Get the current working directory
+        current_dir = os.getcwd()
+
+        def update_app_path(app_path, app_name):
+          with open(app_path, "a") as f:
+            f.write(app_name + "\n")
+
+        if not unique_matches:
+            print("No matching components found in pull request files. Add them to the staging-app.yaml file.")
+        else:
+            for match in unique_matches:
+                for app_name in app.get(match, []):
+                    # Run the argocd command using subprocess
+                    cmd = ["chmod", "0400", f"{current_dir}/argocdconfig"]
+                    cmd_string = " ".join(cmd)
+                    output = subprocess.Popen(cmd_string.split(), stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+                    stdout, stderr = output.communicate()
+                    try:
+                        print(f"trigger the deployment for {app_name}")
+                        cmd = ["argocd", "app", "sync", app_name, "--config", f"{current_dir}/argocdconfig", "--grpc-web"]
+                        cmd_string = " ".join(cmd)
+                        output = subprocess.Popen(cmd_string.split(), stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+                        stdout, stderr = output.communicate()
+                        update_app_path("$(workspaces.infra-deployment-git.path)/component.txt", match + ":" + app_name)
+                        if output.returncode != 0:
+                            print(f"Error occurred: {stderr.decode('utf-8')}")
```

Comment:
You should differentiate this error message from the one throwing the exception. Both messages are quite generic and need to be made a bit more descriptive.